doc/dev_guide/deply_noobaa_on_local_machine.md

```diff
@@ -0,0 +1,131 @@
+# Deploy NooBaa on Local Machine
+There are many ways to deploy NooBaa on k8s on you local machine, this guide refers to 2 options (the difference is in the configurations):
+1. Docker Desktop + Minikube
+2. Rancher Desktop
+
+The guide includes the steps:
+1. Configuration (Docker Desktop + Minikube or Rancher Desktop)
+2. Build images (noobaa-core and noobaa-operator) and optional step to push to a container registry (Quay or Docker Hub).
+3. Install NooBaa using NooBaa CLI.
+
+## 1) Configuration (First Time Only)
+
+### Docker Desktop + Minikube
+#### Change Your Docker Resource Preferences
+Click on docker logo -> preference… -> Resource (left choice in the menu): CPU: 6, Memory: 10 GB, SWAP: 1 GB, Disk: 80 GB
+
+#### Setting Your Minikube Configurations
+a. Start your cluster:
+```bash
+minikube start
+```
+b. Set configurations:
+```bash
+    minikube config set memory 8000
+    minikube config set cpus 5
+```
+
+### Rancher Desktop
+Inside the application -> preferences -> Virtual Machine (left tab) -> Memory: 10 GB, CPU: 6
+
+## General Comment on Working in the Terminal
+We will run the commands in the terminal, you may work with at least two tabs:
+1. For noobaa-core repository
+2. For noobaa-operator repository
+
+In each step, it is mentioned what tab you should use.
+
+## 2) Before Building The Images (Noobaa-Core Tab)
+1) Change the working directory to the local repository (in this example the local repository is inside `SourceCode` directory):
+```bash
+ cd ./SourceCode/noobaa-core
+```
+2) (Minikube only): It is recommended to build all images on the minikube docker daemon. Configure your docker client to use minikube's docker run:
```

Comment:
is there any reason to recommend the noobaa core image build in the docker daemon? what I have seen is if you build the noobaa core in docker daemon it will take more time(only first time in new minikube) since it has to build an image from scratch, so I usually go with system docker and it will be faster because we already build it before.

Reply:
@shirady but for this, local system should have a running docker and the user need to push the image after every build so it looks like its better to go ahead with minikube daemon